middleware/error-handler.js

```diff
@@ -1,10 +1,32 @@
-const { CustomAPIError } = require('../errors')
-const { StatusCodes } = require('http-status-codes')
+const { StatusCodes } = require("http-status-codes")
 const errorHandlerMiddleware = (err, req, res, next) => {
-  if (err instanceof CustomAPIError) {
-    return res.status(err.statusCode).json({ msg: err.message })
+  let customError = {
+    // set default
+    statusCode: err.statusCode || StatusCodes.INTERNAL_SERVER_ERROR,
+    msg: err.message || "Something went wrong try again later",
   }
-  return res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({ err })
+
+  if (err.name === "ValidationError") {
```

Comment:
right now an object is returned with only a message and the status code:
![image](https://user-images.githubusercontent.com/8095975/188292522-6201729a-314f-4f65-b980-327e27733374.png)

should I return an example of how the request should look like?

Reply:
I think the ideal thing to send back is something that matches up with the fields in the form so that server-side validation messages can be rendered on the client. For example:

```javascript
{
  username: [
    "must be at least 5 characters long",
    "cannot contain whitespace characters",
    "cannot contain special characters"
  ],
  password: [
    "cannot be blank",
    "must contain at least one special character",
    " must be at least 10 characters long"
  ]
}
```

So, on the server you would build up an object like this as you run your validation checks, then send it to the client so that the web app consuming your API can tell the end user *why* the form is invalid.

Reply:
That's a great idea! Will work on implementing that!